### PR TITLE
chore(upgrade): v1.4.0 to v1.5.0

### PR DESCRIPTION
**Pull request by enddynayn**
- Upgrade Polkadot-sdk to v.1.5.0.
- Update weights to reflect the new version.
- Modify extrinsic weight test due to a 25% improvement in weights.

For more details, please refer to:

[Release Notes](https://github.com/paritytech/polkadot-sdk/releases/tag/polkadot-v1.5.0)
[Release Analysis](https://forum.polkadot.network/t/polkadot-release-analysis-v1-5-0/5358)

#1813

### CODE REVIEW

**Review by aramikm — APPROVED** [review]
Looks good!

File: runtime/common/src/fee.rs [hunk 1]
```diff
@@ -70,14 +70,16 @@ mod test {
 		assert!(x.max(y) - x.min(y) < MILLICENTS);
 	}
 
-	#[test]
-	fn check_weight() {
-		let p = CENTS / 10;
-		let q = Balance::from(ExtrinsicBaseWeight::get().ref_time());
+	// #[test]
```
Comment (mattheworris) [inline runtime/common/src/fee.rs:73]: Why is this test commented out? Should it be removed?

**Review by mattheworris — APPROVED** [review]
- Read through changes
- Non-blocking comment
lgtm

File: pallets/messages/src/rpc/src/tests/rpc_mock.rs [hunk 2]
```diff
@@ -40,7 +40,7 @@ impl<Block: BlockT> HeaderBackend<Block> for TestApi {
 
 	fn status(
 		&self,
-		_id: <Block as sp_api::BlockT>::Hash,
+		_id: Block::Hash,
```
Comment (saraswatpuneet) [inline pallets/messages/src/rpc/src/tests/rpc_mock.rs:43]: 🥇

File: runtime/common/src/fee.rs (hunk 1 above)
Comment (enddynayn) [reply runtime/common/src/fee.rs:73]: Good question. I am working on getting the new values. The improvements in weights are causing this test to fail.

**Review by claireolmstead — APPROVED** [review]
LGTM!

**Review by saraswatpuneet — APPROVED** [review]
Lgtm!!

File: runtime/common/src/fee.rs (hunk 1 above)
Comment (enddynayn) [reply runtime/common/src/fee.rs]: I wanted to follow up regarding the test we implemented to monitor improvements or regressions in the extrinsic weights. After reviewing it, I have updated the test to reflect the new range of weights based on the improvements observed.

We are not planning to make any changes to our fees at this time. However, please feel free to share your opinion.

**Review by shannonwells — APPROVED** [review]
LGTM

**Pull request merged**